base/linalg/generic.jl

```diff
@@ -1184,9 +1184,36 @@ function logdet(A::AbstractMatrix)
     return d + log(s)
 end
 
+"""
+    promote_leaf_eltypes(itr)
+
+For an (possibly nested) iterable object `itr`, promote the types of leaf
+elements.  Equivalent to `promote_type(typeof(leaf1), typeof(leaf2), ...)`.
+Currently support only numeric leaf elements.
+
+# Example
+
+```jldoctest
+julia> a = [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]]
+3-element Array{Any,1}:
+  Any[1,2,[3,4]]
+ 5.0
+  Any[0+6im,[7.0,8.0]]
+
+julia> promote_leaf_eltypes(a)
+Complex{Float64}
+```
+"""
+typealias NumberArray{T<:Number} AbstractArray{T}
```

Comment:
the docstring should be attached to the method it's documenting, not some unrelated typealias